mindsdb_sdk/agents.py

```diff
@@ -0,0 +1,223 @@
+        :param messages: List of messages to be sent to the agent
+
+        :return: completion from querying the agent
+        """
+        data = self.api.agent_completion(self.project, name, messages)
+        return AgentCompletion(data['message']['content'])
+
+    def create(
+            self,
+            name: str,
+            model: Model,
+            skills: List[Union[Skill, str]] = None,
+            params: dict = None) -> Agent:
+        """
+        Create new agent and return it
+
+        :param name: Name of the agent to be created
+        :param model: Model to be used by the agent
+        :param skills: List of skills to be used by the agent. Currently only 'sql' is supported.
+        :param params: Parameters for the agent
+
+        :return: created agent object
+        """
+        skills = skills or []
+        skill_names = []
+        for skill in skills:
+            if isinstance(skill, str):
+                # Check if skill exists.
+                _ = self.skills.get(skill)
+                skill_names.append(skill)
+                continue
+            # Create the skill if it doesn't exist.
+            _ = self.skills.create(skill.name, skill.type, skill.params)
+            skill_names.append(skill.name)
+
+        data = self.api.create_agent(self.project, name, model.name, skill_names, params)
+        return Agent.from_json(data, self)
+
+    def update(self, name: str, updated_agent: Agent):
+        """
+        Update an agent by name.
+
+        :param name: Name of the agent to be updated
+        :param updated_agent: Agent with updated fields
+
+        :return: updated agent object
+        """
+        updated_skills = set()
+        for skill in updated_agent.skills:
+            if isinstance(skill, str):
+                # Skill must exist.
+                _ = self.skills.get(skill)
+                updated_skills.add(skill)
+                continue
+            try:
+                # Create the skill if it doesn't exist.
+                _ = self.skills.get(skill.name)
+            except HTTPError as e:
+                if e.response.status_code != 404:
+                    raise e
+                # Doesn't exist
+                _ = self.skills.create(skill.name, skill.type, skill.params)
+            updated_skills.add(skill.name)
+
+        existing_agent = self.api.agent(self.project, name)
+        existing_skills = set([s.name for s in existing_agent['skills']])
+        skills_to_add = updated_skills.difference(existing_skills)
+        skills_to_remove = existing_skills.difference(updated_skills)
+        data = self.api.update_agent(
+            self.project,
+            name,
+            updated_agent.name,
+            updated_agent.model_name,
+            list(skills_to_add),
+            list(skills_to_remove),
+            updated_agent.params
+        )
+        return Agent.from_json(data, self)
+
+    def delete(self, name: str):
```

Comment:
could we rename this method to 'drop' for consistency with methods for other entities in SDK API (and for skills too) ?